lua/swenv/api.lua

```diff
@@ -34,104 +34,171 @@ end
-        path = path,
-        source = 'conda',
-      })
-    end
-  end
-
-  -- VENV
-  local paths = scan_dir(venvs_path, { depth = 1, only_dirs = true, silent = true })
-  for _, path in ipairs(paths) do
-    table.insert(venvs, {
-      -- TODO how does one get the name of the file directly?
-      name = Path:new(path):make_relative(venvs_path),
-      path = path,
-      source = 'venv',
-    })
-  end
-
-  return venvs
+    local success, Path = pcall(require, 'plenary.path')
+    if not success then
+        vim.notify('Could not require plenary: ' .. Path, vim.log.levels.WARN)
+        return
+    end
+    local scan_dir = require('plenary.scandir').scan_dir
+
+    local venvs = {}
+
+    -- CONDA
+    local conda_exe = vim.fn.getenv('CONDA_EXE')
+    if conda_exe ~= vim.NIL then
+        local conda_env_path = Path:new(conda_exe):parent():parent() .. '/envs'
+        local conda_paths = scan_dir(conda_env_path, { depth = 1, only_dirs = true, silent = true })
+
+        for _, path in ipairs(conda_paths) do
+            table.insert(venvs, {
+                name = Path:new(path):make_relative(conda_env_path),
+                path = path,
+                source = 'conda',
+            })
+        end
+    end
+
+    -- VENV
+    local paths = scan_dir(venvs_path, { depth = 1, only_dirs = true, silent = true })
+    for _, path in ipairs(paths) do
+        table.insert(venvs, {
+            -- TODO how does one get the name of the file directly?
+            name = Path:new(path):make_relative(venvs_path),
+            path = path,
+            source = 'venv',
+        })
+    end
+
+    return venvs
 end
 
 M.pick_venv = function()
-  vim.ui.select(settings.get_venvs(settings.venvs_path), {
-    prompt = 'Select python venv',
-    format_item = function(item)
-      return string.format('%s (%s) [%s]', item.name, item.path, item.source)
-    end,
-  }, function(choice)
-    if not choice then
-      return
-    end
-    set_venv(choice)
-  end)
+    vim.ui.select(settings.get_venvs(settings.venvs_path), {
+        prompt = 'Select python venv',
+        format_item = function(item)
+            return string.format('%s (%s) [%s]', item.name, item.path, item.source)
+        end,
+    }, function(choice)
+        if not choice then
+            return
+        end
+        set_venv(choice)
+    end)
+end
+
+M.set_venv = function(name)
+    local venvs = settings.get_venvs(settings.venvs_path)
+    local closest_match = best_match(venvs, name)
+    if not closest_match then
+        return
+    end
+    set_venv(closest_match)
+end
+
+M.auto_venv = function()
+    local loaded, project_nvim = pcall(require, "project_nvim.project")
+    local venvs = settings.get_venvs(settings.venvs_path)
+    local project_dir = nil
+    if loaded then
+        project_dir, _ = project_nvim.get_project_root()
+    else
+        print("Error: failed to load the project_nvim.project module")
+        return
+    end
+    if project_dir then
```

Comment:
do you need to check this again here, shouldn't this be not `nil` if `loaded`?

I think it might be simpler to just return early directly if `loaded` if `false`. Then you don't even need to call `get_venvs` unnecessarily.

Reply:
It is necessary, since project.nvim will not always return a path. If no project is located, then there would not be a path.

I have double check'ed it though :)